packages/flutter/lib/src/rendering/editable.dart

```diff
@@ -557,6 +542,55 @@ class RenderEditable extends RenderBox with RelayoutWhenSystemFontsChangeMixin {
     }
   }
 
+  /// Returns the index of the first character after extent.
+  ///
+  /// Setting includeWhitespace to false will only return the index of non-space
+  /// characters.
+  @visibleForTesting
+  static int nextCharacter(int extent, String string, [bool includeWhitespace = true]) {
+    if (extent >= string.length) {
+      return string.length;
+    }
+
+    int count = 0;
+    final Characters remaining = string.characters.skipWhile((String currentString) {
+      if (count <= extent) {
+        count += currentString.length;
+        return true;
+      }
+      if (includeWhitespace) {
+        return false;
+      }
+      return _isWhitespace(currentString.characters.first.toString().codeUnitAt(0));
+    });
+    return string.length - remaining.toString().length;
+  }
+
+  /// Returns the index of the first character before extent.
+  ///
+  /// Setting includeWhitespace to false will only return the index of non-space
+  /// characters.
+  @visibleForTesting
+  static int previousCharacter(int extent, String string, [bool includeWhitespace = true]) {
```

Comment:
This method and the nextCharacter method should make meaning of their extent parameter and return value really clear, i.e. are they string indices or string.character indices. They should make how they handle edge cases clear as well.